src/nvim/extmark.c

```diff
@@ -431,9 +431,32 @@ void u_extmark_copy(buf_T *buf, int l_row, colnr_T l_col, int u_row, colnr_T u_c
       break;
     }
 
-    if (!mt_no_undo(mark)) {
+    bool invalid = mt_invalid(mark);
```

Comment:
would say `was_invalid` as the intent is to remember the old value.

Reply:
Renamed/repurposed to `ExtmarkSavePos.invalidated`, which is only set when a mark was actually invalidated. I think previously the undo header might have been filled with unnecessary `ExtmarkSavePos.invalid == false` marks than necessary.